core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -385,6 +388,225 @@ trait WorkspaceComponent {
+      val substatssql = sql""" LEFT OUTER JOIN (SELECT workspace_id,
+                                                       submission_id,
+                                                       workspace_name,
+                                                       submission_status,
+                                                       workflow_status,
+                                                       MAX(if (workflow_status = '#${WorkflowStatuses.Succeeded.toString}', max_date, null)) AS last_succeeded,
+                                                       MAX(if (workflow_status = '#${WorkflowStatuses.Failed.toString}', max_date, null)) AS last_failed,
+                                                       MAX(submission_active_count) AS running_submission_count,
+                                                       (CASE
+                                                         when MAX(submission_active_count) > 0 then 'Running'
+                                                         when MAX(if (workflow_status = 'Succeeded', max_date, 0)) > MAX(if (workflow_status = 'Failed', max_date, 0)) then 'Succeeded'
+                                                         when MAX(if (workflow_status = 'Succeeded', max_date, 0)) < MAX(if (workflow_status = 'Failed', max_date, 0)) then 'Failed'
+                                                         else null END) AS last_submission_status
+                                                FROM
+                                                    (SELECT w.id AS workspace_id,
+                                                            s.id AS submission_id,
+                                                            w.name AS workspace_name,
+                                                            s.STATUS AS submission_status,
+                                                            wf.STATUS AS workflow_status,
+                                                            MAX(wf.STATUS_LAST_CHANGED) as max_date,
+                                                            SUM(s.STATUS in ('Accepted', 'Evaluating', 'Submitting', 'Submitted', 'Aborting')) AS submission_active_count
+                                                     FROM WORKSPACE w
+                                                              LEFT OUTER JOIN SUBMISSION s on w.id = s.WORKSPACE_ID
+                                                              LEFT OUTER JOIN WORKFLOW wf on s.ID = wf.SUBMISSION_ID
+                                                     GROUP BY s.id, w.id, wf.status) innerS
+                                                GROUP BY workspace_id
+                                                ORDER BY workspace_name) sub_stats ON sub_stats.workspace_id = w.id """
+
+      val fromAndJoins = sql"""
+                         LEFT OUTER JOIN WORKSPACE_ATTRIBUTE wa on ws.id = wa.owner_id
+                         LEFT OUTER JOIN ENTITY e_ref on wa.value_entity_ref = e_ref.id """
```

Comment:
I'm curious about the join to `ENTITY`. Theoretically, a user could store an entity reference in a workspace attribute, and we'd need to retrieve that entity via join. In reality, how many workspaces need this join to retrieve entity info? Less than 1%? Less than 0.01%? You may be able to optimize by doing a two-pass operation: do the simpler query for workspaces, without joining to ENTITY, and get the results. Inspect the results in Scala; if and only if there is a need to query ENTITY, make a second query to resolve the referenced entities.